lib/containers/container_images.pm

```diff
@@ -102,8 +107,8 @@ sub test_opensuse_based_image {
 
     $version = 'Tumbleweed' if ($version =~ /^Staging:/);
 
-    # It is the right version
-    if ($distri eq 'sle') {
+    # SUSEConnect zypper service is supported only on SLE based image on SLE host
+    if (check_os_release('sles', 'ID') && check_os_release('sles', 'ID', "$runtime run --rm $image")) {
```

Comment:
You're calling `check_os_release` quite often, which can be expensive. There is no caching inside the function, but you could use `get_os_release` once instead and do the comparison locally.

Reply:
Done. You can check.